File: src/python/test/readwrite_file.py
class ReadWriteFile:
    def __init__(self, path):
        self.path = path

    def read(self):
        f = open(self.path, 'rb')
        contents = f.read()
        f.close()
        return contents

    def write(self, byte_array):
        f = open(self.path, 'wb')
        f.write(byte_array)
        f.close()

File: src/java/test/ReadWriteFile.kt
package test

import java.io.File

class ReadWriteFile(private val file: File): ReadWrite {

    override val format: String = "file"

    override fun read(): ByteArray {
        return file.readBytes()
    }

    override fun write(data: ByteArray) {
        file.writeBytes(data)
    }
}
